src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala

```diff
@@ -332,7 +332,11 @@ private class AppTaskLauncherActor(
       sender ! MatchedTasks(offer.getId, Seq.empty)
 
     case ActorOfferMatcher.MatchOffer(deadline, offer) =>
-      val newTaskOpt: Option[CreatedTask] = taskFactory.newTask(app, offer, tasksMap.values)
+      val newTaskOpt: Option[CreatedTask] = taskFactory.newTask(
+        app,
+        offer,
+        tasksMap.values.filterNot(_.getStatus.getState == TaskState.TASK_LOST))
```

Comment:
is it worth checking the reason here too? LOST could happen for several reasons right? The goal should be to "not consider terminal tasks when evaluating constraints"